src/translate.rs

```diff
@@ -1162,38 +1180,77 @@ impl FromGlib<i32> for Option<u64> {
     }
 }
 
-/// Translate from a pointer type without taking ownership, transfer: none.
+/// Translate from a pointer type which is annotated with `transfer none`.
+/// The resulting value is referenced at least once, by the bindings.
+///
+/// This is suitable for floating references, which become strong references.
+/// It is also suitable for acquiring non-gobject values, like `gchar*`.
+///
+/// The implementation of this trait should acquire a reference to the value
+/// in a way appropriate to the type, e.g. by referencing or copying.
+/// Values obtained using this trait must be properly released on `drop()`
+/// by the implementing type.
+///
+/// For more information, refer to module level documentation.
 pub trait FromGlibPtrNone<P: Ptr>: Sized {
     unsafe fn from_glib_none(ptr: P) -> Self;
 }
 
-/// Translate from a pointer type taking ownership, transfer: full.
+/// Translate from a pointer type which is annotated with `transfer full`.
+/// This transfers the ownership of the value to the Rust side.
+///
+/// Because ownership can only be transferred if something is already referenced,
+/// this is unsuitable for floating references.
+///
+/// The implementation of this trait should not acquire any references to the value.
+/// Values obtained using this trait must be properly released on `drop()`
+/// by the implementing type.
+///
+/// For more information, refer to module level documentation.
 pub trait FromGlibPtrFull<P: Ptr>: Sized {
     unsafe fn from_glib_full(ptr: P) -> Self;
 }
 
-/// Translate from a pointer type by borrowing. Don't increase the refcount
+/// Translate from a pointer type by borrowing, without affecting the refcount.
+///
+/// The purpose of this trait is to access values inside callbacks
+/// without changing their reference status.
+/// The obtained borrow must not be accessed outside of the scope of the callback.
+/// Care must be taken not to change the reference count of the borrowed value,
```

Comment:
Why? It's perfectly fine (and often required) to call `clone()` on the value to then store it away somewhere. The only thing that must not be done is to take ownership of the value returned by `from_glib_borrow()`

Reply:
Addressed by writing a paragraph on the invariants that the implementation must keep.